lib/charms/alertmanager_k8s/v0/alertmanager.py

```diff
@@ -86,37 +84,27 @@ def __init__(self, charm: CharmBase, relation_name: str, consumes: dict, multi:
             self.charm.on[self._consumer_relation_name].relation_broken, self._on_relation_broken
         )
 
-        self._stored.set_default(alertmanagers={})
-
     def _on_relation_changed(self, event: ops.charm.RelationChangedEvent):
-        """This hook stores locally the address of the newly-joined alertmanager.
-        This is needed for consumers such as prometheus, which should be aware of all alertmanager
-        instances.
-        """
+        """This hook notifies the charm that there may have been changes to the cluster"""
         if event.unit:  # event.unit may be `None` in the case of app data change
-            # Save locally the public IP address of the alertmanager unit
-            if address := event.relation.data[event.unit].get("public_address"):
-                # TODO consider storing in unit data instead of StoredState
-                self._stored.alertmanagers[event.unit.name] = address
-
-                # inform consumer about the change
-                self.cluster_changed.emit()
+            # inform consumer about the change
+            self.cluster_changed.emit()
 
     def get_cluster_info(self) -> List[str]:
         """Returns a list of ip addresses of all the alertmanager units"""
-        return sorted(list(self._stored.alertmanagers.values()))
+        alertmanagers = []
+        if not (relation := self.charm.model.get_relation(self._consumer_relation_name)):
+            return alertmanagers
+        for unit in relation.units:
+            if address := relation.data[unit].get("public_address"):
+                alertmanagers.append(address)
```

Comment:
I would left one empty  line between the if structure and the for, but it's a matter of taste ;-)